src/coreclr/nativeaot/Runtime/gctoclreventsink.cpp

```diff
@@ -11,17 +11,21 @@ void GCToCLREventSink::FireDynamicEvent(const char* eventName, void* payload, ui
 {
     LIMITED_METHOD_CONTRACT;
 
-#ifndef FEATURE_NATIVEAOT
     const size_t EventNameMaxSize = 255;
 
     WCHAR wideEventName[EventNameMaxSize];
-    if (MultiByteToWideChar(CP_ACP, 0, eventName, -1, wideEventName, EventNameMaxSize) == 0)
+    int i = 0;
+    while (true)
```

Comment:
What is the recommended way for this?

Reply:
I think what you have here is fine. I would update the CoreCLR copy to do the same. `CP_ACP` does not make sense as the encoding to use here.

(Alternatively, this can use `minipal_convert_utf8_to_utf16` but it is probably more troubles than it is worth it to plumb it through.)